backend/frontend/views.py

```diff
@@ -295,6 +297,77 @@ def action_start_card(request, card_id):
     )
 
 
+def user_can_request_review(logged_in_user):
+    request = get_current_request()
+    card_id = request.resolver_match.kwargs.get("card_id")
+
+    card = get_object_or_404(AgileCard, pk=card_id)
+    card_assignees = card.assignees.all()
+
+    if logged_in_user in card_assignees:
+        return True
+
+    card_teams = card.get_teams()
+    checker = ObjectPermissionChecker(logged_in_user)
+    checker.prefetch_perms(card_teams)
+
+    return any(
+        (checker.has_perm(Team.PERMISSION_MANAGE_CARDS, team) for team in card_teams)
+    )
+
+
+@csrf_exempt
+@user_passes_test_or_forbidden(user_can_request_review)
+def action_request_review(request, card_id):
+    """The card is in progress or review feedback and the user has chosen to request review"""
+    card = get_object_or_404(AgileCard, id=card_id)
+
+    if len(helpers.agile_card_reviews_outstanding(request.user)):
+        return render(
+            request,
+            "frontend/user/board/js_exec_action_show_card_alert.html",
+            {
+                "card": card,
+                "message": "Please make sure you have reviewed assigned cards.",
+                "alert_type": "error",
+            },
+        )
+
+    if len(helpers.pull_request_reviews_outstanding(request.user)):
+        return render(
+            request,
+            "frontend/user/board/js_exec_action_show_card_alert.html",
+            {
+                "card": card,
+                "message": "Please make sure you have reviewed assigned pull requests.",
+                "alert_type": "error",
+            },
+        )
+
+    if card.recruit_project:
+        card.recruit_project.request_review(force_timestamp=timezone.now())
+    elif card.topic_progress:
```

Comment:
The request review button should not show up for topics. If there is a topic_progress instance it would be an error.

For topics you should just be able to say that they are finished, it's a different button